src/Stache/Stores/Store.php

```diff
@@ -27,6 +27,57 @@ abstract class Store
     protected $shouldCacheFileItems = false;
     protected $modified;
     protected $keys;
+    protected $identifiedBy = 'id';
+
+    protected function resolveFromIndex($keys, $column)
+    {
+        return $this->resolveIndex($column)
+            ->load()
+            ->items()
+            ->where(function ($value, $key) use (&$keys) {
+                return $keys->has($key);
+            });
+    }
+
+    private function isValidKey($value)
+    {
+        if (is_string($value) || is_int($value)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function getFromIndex($keys, $column, $key = null)
+    {
+        if ($column === $this->identifiedBy && $key === null) {
+            return $keys;
+        }
+
+        $keys = $keys->flip();
+        $values = $this->resolveFromIndex($keys, $column);
+
+        if ($key === null) {
+            return $values->values();
+        }
+
+        $keyValues = $this->resolveFromIndex($keys, $key);
+        $newValues = [];
+
+        foreach ($keys->keys() as $keyValue) {
+            $newKeyValue = $keyValues[$keyValue] ?? null;
+
+            if (! $this->isValidKey($newKeyValue)) {
+                continue;
+            }
```

Comment:
This is never reached.

Reply:
Need to request something like a Bard field for the `key`. Will add test coverage 👍

Reply:
I've removed this after thinking about it some more. If someone requests invalid data getting an empty result is weird. At least the error will help troubleshoot what's actually going on